File: __Viejo/funciones/generales_multiUsuario.vb
Imports System.Data.SqlClient

Module generales_multiUsuario

    Public Function borrar_tabla_pedidos_temporales(ByVal idUsuario As Integer, ByVal idUnico As String) As Byte
        abrirdb(serversql, basedb, usuariodb, passdb)

        Dim mytrans As SqlTransaction
        Dim Comando As New SqlCommand
        Dim sqlstr As String

        mytrans = CN.BeginTransaction()

        Try
            sqlstr = "DELETE tmppedidos_items WHERE id_usuario = '" + idUsuario.ToString + "' AND id_unico = '" + idUnico + "'"
            Comando = New SqlCommand(sqlstr, CN) With {.Transaction = mytrans}
            Comando.ExecuteNonQuery()

            mytrans.Commit()
            cerrardb()

            Return True

        Catch ex As Exception
            MsgBox(ex.Message)
            cerrardb()
            Return False
        End Try
    End Function

    Public Function borrar_tabla_pedidos_temporales(ByVal idUsuario As String) As Byte
        abrirdb(serversql, basedb, usuariodb, passdb)

        Dim mytrans As SqlTransaction
        Dim Comando As New SqlCommand
        Dim sqlstr As String

        mytrans = CN.BeginTransaction()

        Try
            sqlstr = "DELETE tmppedidos_items WHERE id_usuario = '" + idUsuario + "'"
            Comando = New SqlCommand(sqlstr, CN) With {.Transaction = mytrans}
            Comando.ExecuteNonQuery()

            mytrans.Commit()
            cerrardb()

            Return True

        Catch ex As Exception
            MsgBox(ex.Message)
            cerrardb()
            Return False
        End Try
    End Function

    Public Function Borrar_tabla_segun_usuario(ByVal tbl As String, ByVal id_usuario As Integer) As Byte
        Dim sqlstr As String
        abrirdb(serversql, basedb, usuariodb, passdb)

        Dim mytrans As SqlTransaction
        Dim Comando As New SqlClient.SqlCommand

        mytrans = CN.BeginTransaction()

        Try
            'sqlstr = "DELETE FROM " + tbl + "; DBCC CHECKIDENT ('" + basedb + ".dbo." + tbl + "',RESEED, 0)"
            sqlstr = "DELETE FROM " + tbl + " WHERE id_usuario = '" + id_usuario.ToString + "'; DBCC CHECKIDENT ('" + tbl + "',RESEED, 0)"

            Comando = New SqlClient.SqlCommand(sqlstr, CN)

            Comando.Transaction = mytrans
            Comando.ExecuteNonQuery()

            mytrans.Commit()
            cerrardb()

            Return True

        Catch ex As Exception
            MsgBox(ex.Message)
            cerrardb()
            Return False
        End Try
    End Function

    Public Sub borrarTmpProduccion(ByVal id_usuario As Integer)
        Borrar_tabla_segun_usuario("tmpproduccion_asocItems", id_usuario)
        Borrar_tabla_segun_usuario("tmpproduccion_items", id_usuario)
    End Sub

    Public Function Generar_ID_Unico() As String
        Dim idUnico As String

        idUnico = ejecutarSQLconResultado("SELECT CONVERT(NVARCHAR(100), NEWID())")

        Return idUnico
    End Function
End Module

File: funciones/generales_multiUsuario.cs
using System;
using System.Data;
using System.Linq;
namespace Centrex.Funciones
{
    public static class generales_multiUsuario
    {
        // ************************************* FUNCIONES MULTI-USUARIO CON ENTITY FRAMEWORK *****************************

        /// <summary>
        // ************************************* FUNCIONES MULTI-USUARIO CON ENTITY FRAMEWORK *****************************

        /// <summary>
        /// Borra tabla de pedidos temporales según usuario - Migrado a EF
        /// </summary>
        public static byte borrar_tabla_pedidos_temporales(int idUsuario)
        {
            try
            {
                using (var context = new CentrexDbContext())
                {

                    var itemsTemp = context.TmpPedidoItemEntity.Where(t => t.IdUsuario == idUsuario).ToList();
                    if (itemsTemp.Count == 0)
                    {
                        return Conversions.ToByte(true);
                    }

                    context.TmpPedidoItemEntity.RemoveRange(itemsTemp);
                    context.SaveChanges();
                    return Conversions.ToByte(true);
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(ex.Message);
                return Conversions.ToByte(false);
            }
        }

        //// NOTA: Función duplicada en produccion.vb, esta versión está comentada para evitar ambigüedad
        //public static void borrarTmpProduccionGeneral(int id_usuario)
        //{
        //    Borrar_tabla_segun_usuario("tmpproduccion_asocItems", id_usuario);
        //    Borrar_tabla_segun_usuario("tmpproduccion_items", id_usuario);
        //}

        /// <summary>
        /// Borra tabla según usuario - Migrado a EF
        /// </summary>
        public static bool Borrar_tabla_segun_usuario(string tbl, int id_usuario)
        {
            try
            {
                using (var context = new CentrexDbContext())
                using (var dbTrans = context.Database.BeginTransaction())
                {
                    try
                    {
                        if (string.IsNullOrWhiteSpace(tbl) || tbl.Any(c => !char.IsLetterOrDigit(c) && c != '_'))
                            throw new ArgumentException("Nombre de tabla inválido: " + tbl);

                        string sqlDelete = $"DELETE FROM [{tbl}] WHERE id_usuario = {id_usuario};";
                        context.Database.ExecuteSqlRaw(sqlDelete);

                        string sqlReseed = $"DBCC CHECKIDENT ('[{tbl}]', RESEED, 0);";
                        context.Database.ExecuteSqlRaw(sqlReseed);

                        dbTrans.Commit();
                        return true;
                    }
                    catch (Exception innerEx)
                    {
                        dbTrans.Rollback();
                        Interaction.MsgBox(
                            $"Error al limpiar tabla '{tbl}': {innerEx.Message}",
                            MsgBoxStyle.Critical,
                            "Centrex"
                        );
                        return false;
                    }
                }
            }
            catch (Exception ex)
            {
                Interaction.MsgBox(
                    $"Error general en Borrar_tabla_segun_usuario: {ex.Message}",
                    MsgBoxStyle.Critical,
                    "Centrex"
                );
                return false;
            }
        }

        /// <summary>
        /// Borra registros temporales de producción para un usuario específico
        /// </summary>
        public static void borrarTmpProduccion(int id_usuario)
        {
            try
            {
                using (var context = new CentrexDbContext())
                {
                    var tmpItems = context.TmpProduccionItemEntity.Where(t => t.IdUsuario == id_usuario).ToList();
                    if (tmpItems.Count > 0)
                        context.TmpProduccionItemEntity.RemoveRange(tmpItems);

                    var tmpAsocItems = context.TmpProduccionAsocItemEntity.Where(t => t.IdUsuario == id_usuario).ToList();
                    if (tmpAsocItems.Count > 0)
                        context.TmpProduccionAsocItemEntity.RemoveRange(tmpAsocItems);

                    context.SaveChanges();
                }
            }
            catch
            {
                // Intencionalmente silencioso para mantener comportamiento legacy
            }
        }

        /// <summary>
        /// Genera ID único usando NEWID() de SQL Server
        /// </summary>
        public static Guid Generar_ID_Unico()
        {
            try
            {
                // 🔹 Generar un GUID nativo en formato estándar
                return Guid.NewGuid();
            }
            catch (Exception)
            {
                // 🔸 En caso de error (extremadamente raro), devolver un GUID vacío
                return Guid.Empty;
            }
        }
    }
}